### PR TITLE
Added TF TransfoXL Sequence Classification

### LINKED ISSUE #7623: Implement PyTorch and/or TensorFlow sequence classification architectures for causal language models
# 🚀 Feature request

The architecture `GPT2ForSequenceClassification` was added in #7501 in PyTorch. It would be great to have it in TensorFlow (cf. issues #7622), but it would also be great to have it for other causal models: ~OpenAI GPT~, ~CTRL~ (PR opened @elk-cloner), ~TransfoXL~ (PR opened @spatil6)

Below is a list of items to follow to make sure the integration of such an architecture is complete:

- Implement `XXXForSequenceClassification` in `modeling_xxx.py` or `TFXXXForSequenceClassification` in `modeling_tf_xxx.py
- Test that architecture in `tests/test_modeling_xxx.py` or `tests/test_modeling_tf_xxx.py`
- Add that architecture to `__init__.py` and ` docs/source/model_doc/xxx.rst`.

Taking a look at the code changes in #7501 would be a good start.

A very good first issue to get acquainted with the library and its architectures!

### PR DESCRIPTION
**Pull request by spatil6**
This PR implements Sequence classification for TF TransfoXL model.
TFTransfoXLForSequenceClassification uses the last token in order to do the classification, as other causal models (e.g. GPT-1 ,GPT-2) do.

Fixes #7623

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

 @LysandreJik @jplu

### CODE REVIEW

**Review by LysandreJik — CHANGES_REQUESTED** [review]
This LGTM. @jplu if you have nothing to say, feel free to merge the PR.

**Review by LysandreJik — APPROVED** [review]
Meant to approve, finger slipped.

File: tests/test_modeling_tf_transfo_xl.py [hunk 1]
```diff
@@ -131,6 +141,12 @@ def create_and_check_transfo_xl_lm_head(self, config, input_ids_1, input_ids_2,
             [(self.mem_len, self.batch_size, self.hidden_size)] * self.num_hidden_layers,
         )
 
+    def create_and_check_transfo_xl_for_sequence_classification(self, config, input_ids_1, input_ids_2, lm_labels):
+        config.num_labels = self.num_labels
```
Comment (spatil6) [inline tests/test_modeling_tf_transfo_xl.py:145]: @LysandreJik I have defined 'num_labels' here.
Now,I'll move it to 'prepare_config_and_inputs'

File: tests/test_modeling_tf_transfo_xl.py [hunk 2]
```diff
@@ -77,6 +85,8 @@ def prepare_config_and_inputs(self):
             div_val=self.div_val,
             n_layer=self.num_hidden_layers,
             eos_token_id=self.eos_token_id,
+            pad_token_id=self.vocab_size - 1,
+            init_range=self.init_range,
         )
```
[the remark this answers is not in the document]
Comment (spatil6) [reply tests/test_modeling_tf_transfo_xl.py]: done

**Review by jplu — APPROVED** [review]
LGTM

**Pull request merged**